src/__test__/generate-test-data.ts

```diff
@@ -38,12 +46,12 @@ export function generateLayout(nDims, nMeas, nRows, qColumnOrder = [], qGrandTot
 
   return {
     qHyperCube: {
-      qDimensionInfo,
-      qMeasureInfo,
+      qDimensionInfo: qDimensionInfo as unknown as EngineAPI.INxDimensionInfo[],
+      qMeasureInfo: qMeasureInfo as unknown as EngineAPI.INxMeasureInfo[],
       qGrandTotalRow,
       qColumnOrder,
       qSize: { qcx: nDims + nMeas, qcy: nRows },
-    },
+    } as unknown as HyperCube,
```

Comment:
Yeah the thing is that we only want to mock a subset of everything that is on some of these types. And I get an error if I just do ` as HyperCube`. Already doing this in some tests. The option would be, as you say, to create a new type, but that would make the typing less strict, since we would have `theRealType | theMockedType`. Or we mock it properly, but it does not add any real value